src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -310,6 +321,49 @@ def checkm2(genome_ids, work_dir, source_data_dir, debug, program_threads, batch
     return failed_ids
 
 
+def microtrait(genome_ids, work_dir, source_data_dir, debug, program_threads, node_id, source_file_ext):
+    failed_ids, size = list(), len(genome_ids)
+    print(f'Start executing MicroTrait for {len(genome_ids)} genomes')
+
+    batch_dir = _create_batch_dir(work_dir, 'series', size, node_id)
+    tool_genome_id_map, source_genome_file_map = dict(), dict()
+    # retrieve genomic.fna.gz files
+    fna_files = list()
+    for genome_id in genome_ids:
+        genome_file = _find_genome_file(genome_id, source_file_ext, source_data_dir,
+                                        exclude_file_name_substr=['cds_from', 'rna_from', 'ERR'],
+                                        expected_file_count=1)
+
+        if genome_file:
+            tool_genome_id_map.update(_map_tool_id_to_genome_id('microtrait',
+                                                                genome_id,
+                                                                os.path.basename(genome_file[0]),
+                                                                source_file_ext))
+            source_genome_file_map[genome_id] = genome_file[0]
+            fna_files.append(str(genome_file[0]))
```

Comment:
They are some duplicate codes, but I really don't see how we can future abstract. Yes, all those 3 tools are fetching genome files, but gtdb-tk needs a master file that has all genome and genome files info, checkm2 needs to pass that file list to the commend line. Microtrait and checkm2 share some similarities. But it might get changed down the road. I can probably do a function that takes genome_ids and return a dict with genome_id and file_path. But then all those tool functions also need to go over the whole dict and process genome files to suit their needs.  Do you think that worth the refactor?

Reply:
I'm talking specifically about lines 332-343 above - I'm thinking you're talking about the whole function? AFAICT those lines are almost identical

Reply:
👍 refactored

Reply:
:+1: I'll add any comments about the refactor to a new review